### ISSUE TITLE
Outputs of type "Directory" to not get copied/moved to the output directory.

### ISSUE THREAD

**Issue by drtconway** [PMCC-BioinformaticsCore/janis-core#13]
Hi Janis,

I may have misunderstood, well, lots of things. So this may be a bug in my comprehension, not a problem with Janis.

Consider the following workflow which creates a working directory for putting a group of files used by a tool in. The command for doing that takes a Filename and returns a Directory. The workflow then executes a step that takes that directory and creates one or more files in that directory. If that step returns a list of files created in that directory, everything works fine. If instead it returns the directory, so as to remain agnostic to exactly what files got put in that directory, the workflow succeeds, but does not copy anything to the output directory.

Test case below. See the commented sections for the succeeding and failing versions.

The invocation I used was:
```bash
janis run -o qux dirout.py --dirname foo --filename bar
```

```python
from typing import List
from janis_core import Directory, File, Filename, InputSelector, PythonTool, TOutput, WorkflowBuilder
from janis_core.tool.commandtool import CommandToolBuilder, ToolInput, ToolOutput

MkdirP = CommandToolBuilder(
    tool="mkdirp",
    base_command=["mkdir", "-p"],
    version="0.1.0",
    container="ubuntu:latest",
    inputs=[
        ToolInput('path', Filename, position=1, doc='directory path to create')
    ],
    outputs=[
        ToolOutput('out', Directory, InputSelector('path'))
    ]
)

class TouchInDir1(PythonTool):
    @staticmethod
    def code_block(dirname: Directory, filename: Filename):
        import os.path

        if not os.path.isdir(dirname):
            raise Exception(f"directory '{dirname}' does not exist.")

        pth = os.path.join(dirname, filename)
        with open(pth, 'w') as f:
            pass
        return {"out": pth}

    # NB this version returns a File.
    def outputs(self) -> List[TOutput]:
        return [TOutput("out", File)]

    def id(self) -> str:
        return "TouchInDir1"

    def friendly_name(self):
        return "touch a file in a directory."

    def version(self):
        return "v0.1.0"

    def tool_provider(self):
        return "Peter MacCallum Cancer Centre"

    def bind_metadata(self):
        self.metadata.creator = "Thomas Conway"
        self.metadata.dateCreated = "2020-03-30"
        self.metadata.version = "0.1.0"

class TouchInDir2(PythonTool):
    @staticmethod
    def code_block(dirname: Directory, filename: Filename):
        import os.path

        if not os.path.isdir(dirname):
            raise Exception(f"directory '{dirname}' does not exist.")

        pth = os.path.join(dirname, filename)
        with open(pth, 'w') as f:
            pass
        return {"out": dirname}

    # NB this version returns a Directory
    def outputs(self) -> List[TOutput]:
        return [TOutput("out", Directory)]

    def id(self) -> str:
        return "TouchInDir2"

    def friendly_name(self):
        return "touch a file in a directory."

    def version(self):
        return "v0.1.0"

    def tool_provider(self):
        return "Peter MacCallum Cancer Centre"

    def bind_metadata(self):
        self.metadata.creator = "Thomas Conway"
        self.metadata.dateCreated = "2020-03-30"
        self.metadata.version = "0.1.0"

w = WorkflowBuilder("mdirAndTouch")
w.input("dirname", Filename)
w.input("filename", Filename)
w.step("mkdir", MkdirP(path=w.dirname))

# This works
#w.step("touch", TouchInDir1(dirname=w.mkdir, filename=w.filename))

# This doesn't
w.step("touch", TouchInDir2(dirname=w.mkdir, filename=w.filename))

w.output("out", source=w.touch)
```

**Comment by drtconway** [comment]
Hmm. Ok. I've just been reading the WDL specification, and I see it doesn't support Directory as a type, but only File, so there's no way to return the directory, and perhaps the only way is to return all the individual files.

**Comment by illusional** [comment]
Hey @drtconway, we use the [Development spec](https://github.com/openwdl/wdl/blob/master/versions/development/SPEC.md) of WDL, which supports directory. 

This is an issue, I'm pretty sure I just haven't implemented the recursive copying yet.

**Comment by illusional** [comment]
Hi Tom, in https://github.com/PMCC-BioinformaticsCore/janis-assistant/pull/11 I've implemented the output copying for directories. 

Your use case brings up an interesting problem, depending on the engine (CWLTool, Toil, Cromwell), the directories (and inputs in general) may not be inherently modifiable without providing an additional tag. I don't have a solution for you yet (maybe some `Writable` annotation), but keep me posted if you have any problems.

This is released in [`v0.9.13`](https://github.com/PMCC-BioinformaticsCore/janis-assistant/releases).
